### PR TITLE
Fix index operator against Series and Frame to use iloc conditionally

### LINKED ISSUE #287: Cannot select [] with dates as index
Run the query:
```
kdf[0:3]
```
and get the error:
```
AttributeError: module 'pyarrow.pandas_compat' has no attribute 'make_datetimetz'
```

To recreate `kdf`:
```
dates = pd.date_range('20130101', periods=6)
pdf = pd.DataFrame(np.random.randn(6,4), index=dates, columns=list('ABCD'))
kdf = ks.from_pandas(pdf)
```

### PR DESCRIPTION
**Pull request by HyukjinKwon**
Looks pandas conditionally uses `iloc` approach when the input is `int` against index operators in Series and Frame.

```python
import databricks.koalas as ks
import pandas as pd
import numpy as np

dates = pd.date_range('20130101', periods=6)
pdf = pd.DataFrame(np.random.randn(6,4), index=dates, columns=list('ABCD'))
kdf = ks.from_pandas(pdf)
kdf[0:3]
```

```
                   A         B         C         D
2013-01-01  2.143975 -2.069608 -0.013988 -0.311598
2013-01-02 -0.448577  0.076479 -0.356269  0.965872
2013-01-03  0.203477  0.738288 -1.453629 -0.146019
```

Resolves #287

### CODE REVIEW

File: databricks/koalas/tests/test_indexing.py [hunk 1]
```diff
@@ -13,8 +13,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
-
```
Comment (itholic) [inline databricks/koalas/tests/test_indexing.py]: nit: maybe this change is not necessary?

**Comment by itholic** [comment]
LGTM

**Comment by HyukjinKwon** [comment]
oops, sure. I missed the line from the issue.

**Comment by HyukjinKwon** [comment]
Merged! thanks, @ueshin and @itholic

**Pull request merged**